pymc3/distributions/bound.py

```diff
@@ -60,153 +64,39 @@ def logp(self, value):
         -------
         TensorVariable
         """
-        logp = self._wrapped.logp(value)
-        bounds = []
-        if self.lower is not None:
-            bounds.append(value >= self.lower)
-        if self.upper is not None:
-            bounds.append(value <= self.upper)
-        if len(bounds) > 0:
-            return bound(logp, *bounds)
-        else:
-            return logp
-
-    def _random(self, lower, upper, point=None, size=None):
-        lower = np.asarray(lower)
-        upper = np.asarray(upper)
-        if lower.size > 1 or upper.size > 1:
-            raise ValueError(
-                "Drawing samples from distributions with array-valued bounds is not supported."
-            )
-        total_size = np.prod(size).astype(np.int)
-        samples = []
-        s = 0
-        while s < total_size:
-            sample = np.atleast_1d(self._wrapped.random(point=point, size=total_size)).flatten()
-
-            select = sample[np.logical_and(sample >= lower, sample <= upper)]
-            samples.append(select)
-            s += len(select)
-        if size is not None:
-            return np.reshape(np.concatenate(samples)[:total_size], size)
-        else:
-            return samples[0]
+        logp = _logp(distribution.owner.op, value, {}, *distribution.owner.inputs[3:])
+        bounds = [(value >= lower), (value <= upper)]
+
+        return bound(logp, *bounds)
+
+
+class _ContinuousBounded(BoundedContinuous):
+    rv_op = boundrv
+
+    def __new__(cls, *args, **kwargs):
+        cls.bound_args_indices = [1, 2]
+        return super().__new__(cls, *args, **kwargs)
```

Comment:
Does this work?

```suggestion
    rv_op = boundrv
    cls.bound_args_indices = [-2, -1]
```

Reply:
Yes , that's probably a better way to do it.